sdk/python/feast/infra/offline_stores/file.py

```diff
@@ -208,6 +211,74 @@ def evaluate_historical_retrieval():
         job = FileRetrievalJob(evaluation_function=evaluate_historical_retrieval)
         return job
 
+    @staticmethod
+    def get_historical_features_by_view(
+        config: RepoConfig,
+        feature_views: List[FeatureView],
+        feature_refs: List[str],
+        entity_df: Union[pd.DataFrame, str],
+        registry: Registry,
+        project: str,
+        start_date: Optional[datetime],
+        end_date: Optional[datetime],
+    ):
+        if not (isinstance(entity_df, pd.DataFrame) or isinstance(entity_df, str)):
+            raise ValueError(
+                f"Please provide an entity_df of type {type(pd.DataFrame)} instead of type {type(entity_df)}"
+                f"Please provide an entity_df of type {type(pd.DataFrame)} or {type(str)} instead of type {type(entity_df)}"
+            )
+
+        if isinstance(entity_df, str):
+            for feature_view in feature_views:
+                if feature_view.name == entity_df:
+                    join_keys = []
+                    for entity_name in feature_view.entities:
+                        entity = registry.get_entity(entity_name, project)
+                        join_keys.append(entity.join_key)
+
+                    columns = join_keys + [feature_view.input.event_timestamp_column]
+                    entity_df = pyarrow.parquet.read_table(
+                        feature_view.input.path, columns=columns
+                    ).to_pandas()
+                    entity_df.rename(
+                        columns={
+                            feature_view.input.event_timestamp_column: DEFAULT_ENTITY_DF_EVENT_TIMESTAMP_COL
+                        },
+                        inplace=True,
+                    )
+                    break
+            else:
+                raise FeatureViewNotFoundException(entity_df)
+
+        entity_df_event_timestamp_col = DEFAULT_ENTITY_DF_EVENT_TIMESTAMP_COL  # local modifiable copy of global variable
+        if entity_df_event_timestamp_col not in entity_df.columns:
```

Comment:
The only time we "visit" this code is if the `entity_df` provided is a `pd.DataFrame` - should we exclude this from happening? Or should the method be renamed.